pkg/R/sparkR.R

```diff
@@ -36,14 +36,17 @@ sparkR.stop <- function(env) {
   if (exists(".sparkRjsc", envir = env)) {
     sc <- get(".sparkRjsc", envir = env)
     callJMethod(sc, "stop")
+    rm(".sparkRjsc", envir = env)
   }
 
```

Comment:
@oscaroboto - So technically we would like both to exist at the same time, but that doesn't always happen to be the case. Let me explain a bug that this code path was fixing (and this will also explain why connExists looks the way it does)

When `sparkR.stop` is called we want to do two things (a) stop the SparkContext and (b) stop the backend handler (Actually I am not that sure about (a), but lets postpone that for now). The problem is that to do either of those we need an active connection to the backend

- On exit from the R shell when one doesn't save the workspace, we usually have an active connection, so everything works fine i.e. `.sparkRCon` and `.sparkRjsc` both exit and the connection is valid. If you remove `.sparkRCon` from the env, this case will be idempotent i.e. you can call `sparkR.stop` many times without worry.
- On exit from the R shell (or RStudio) when one saves the workspace, R closes all active connections before calling the finalizer. So in this case we have a value in `.sparkRCon` but that connection is not valid. This is the main reason we have `connExists` right now is to catch an exception when we call `isOpen`.  Since we definitely want to stop the backend, we re-open a new connection if one does not exist. Now to make this idempotent what we need to do is to use `tryCatch` around the call to `connectBackend`. That way if the backend has already exited and we can't connect to it, we can just return from this function.
- Finally the `sleep(1)` should be only required in cases where we call `stopBackend` - so your call on that is correct. 

In summary I think the only thing that needs to change as a part of this PR is adding a `tryCatch` to `connectBackend` in this function and of course the `rm(.sparkRJsc)` above. Let me know if my analysis missed some use case.